examples/10_HasegawaWakatani/10_hasegawa_wakatani.cpp

```diff
@@ -0,0 +1,734 @@
+  // \param init_val [in] Initial value of the variables
+  Variables(const Grid& grid, double init_val = 0.001) {
+    auto rand_engine      = std::mt19937(0);
+    auto rand_dist        = std::uniform_real_distribution<double>(0.0, 1.0);
+    constexpr int nb_vars = 2;
+    const int nkx2 = grid.m_ksq.extent(1), nkyh = grid.m_ksq.extent(0);
+    m_fk = View3D<Kokkos::complex<double>>("fk", nb_vars, nkyh, nkx2);
+    m_pk = View2D<Kokkos::complex<double>>("pk", nkyh, nkx2);
+
+    const Kokkos::complex<double> I(0.0, 1.0);  // Imaginary unit
+    auto h_fk = Kokkos::create_mirror_view(m_fk);
+    auto h_ksq =
+        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), grid.m_ksq);
+    for (int iky = 0; iky < nkyh; iky++) {
+      for (int ikx = 0; ikx < nkx2; ikx++) {
+        double random_number = rand_dist(rand_engine);
+        h_fk(0, iky, ikx)    = init_val / (1.0 + h_ksq(iky, ikx)) *
+                            Kokkos::exp(I * 2.0 * M_PI * random_number);
+        h_fk(1, iky, ikx) = -h_fk(0, iky, ikx) * h_ksq(iky, ikx);
+      }
+    }
+    Kokkos::deep_copy(m_fk, h_fk);
+  }
+};
+
+// \brief A class to represent the 4th order Runge-Kutta method for solving ODE
+// dy/dt = f(t, y) by
+// y^{n+1} = y^{n} + (k1 + 2*k2 + 2*k3 + k4)/6
+// t^{n+1} = t^{n} + h
+// where h is a time step and
+// k1 = f(t^{n}      , y^{n}     ) * h
+// k2 = f(t^{n} + h/2, y^{n}+k1/2) * h
+// k3 = f(t^{n} + h/2, y^{n}+k2/2) * h
+// k4 = f(t^{n} + h  , y^{n}+k3  ) * h
+//
+// \tparam ViewType The type of the view
+template <typename ViewType>
+class RK4th {
+  using value_type     = typename ViewType::non_const_value_type;
+  using float_type     = KokkosFFT::Impl::base_floating_point_type<value_type>;
+  using BufferViewType = View1D<value_type>;
+
+  //! Order of the Runge-Kutta method
+  const int m_order = 4;
+
+  //! Time step size
+  const float_type m_h;
+
+  //! Size of the input View after flattening
+  std::size_t m_array_size;
+
+  //! Buffer views for intermediate results
+  BufferViewType m_y, m_k1, m_k2, m_k3;
+
+ public:
+  // \brief Constructor of a RK4th class
+  // \param y [in] The variable to be solved
+  // \param h [in] Time step
+  RK4th(const ViewType& y, float_type h) : m_h(h) {
+    m_array_size = y.size();
+    m_y          = BufferViewType("y", m_array_size);
+    m_k1         = BufferViewType("k1", m_array_size);
+    m_k2         = BufferViewType("k2", m_array_size);
+    m_k3         = BufferViewType("k3", m_array_size);
+  }
+
+  auto order() { return m_order; }
+
+  // \brief Advances the solution by one step using the Runge-Kutta method.
+  // \param dydt [in] The right-hand side of the ODE
+  // \param y [in] The current solution.
+  // \param step [in] The current step (0, 1, 2, or 3)
+  void advance(const ViewType& dydt, const ViewType& y, int step) {
+    auto h                = m_h;
+    auto* y_data          = y.data();
+    auto* y_copy_data     = m_y.data();
+    const auto* dydt_data = dydt.data();
```

Comment:
Why do we need to work with pointers ? I don't feel we should show this in the blog

Reply:
I use pointers for two reasons: performance and simplicity.
Because the `y` and `dydt` can be multidimensional, we need MDRange for computations. However, for time advances, the dimensionally of variables does not matter

Reply:
In case the user passes a LayoutStride we cannot ensure the result to be correct, right ?

I suggest the caller/user of this class is responsible to construct and pass 1D Views so that we can work with the function call operator.